src/com/amplitude/api/Utils.java

```diff
@@ -103,4 +106,20 @@ static String normalizeInstanceName(String instance) {
         }
         return instance.toLowerCase();
     }
+
+    static SharedPreferences getAmplitudeSharedPreferences(Context context, String instanceName) {
+        assert !isEmptyString(instanceName);
+        String prefName = Constants.PACKAGE_NAME + "." + instanceName + "." + context.getPackageName();
+        return context.getSharedPreferences(prefName, Context.MODE_PRIVATE | Context.MODE_MULTI_PROCESS);
```

Comment:
I think the `Context.MODE_MULTI_PROCESS` syncs the shared preferences across multiple processes, but probably not completely necessary

Reply:
This is also a single instance shared across all clients so we don't need to cache a pointer to it anywhere (or write our own wrapper)